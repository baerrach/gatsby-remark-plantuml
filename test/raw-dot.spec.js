const testRemarkPlugin = require(`./test-remark-plugin`)

describe(`dot code`, () => {
  beforeEach(() => {
    jest.resetModules()
  })

  it(`renders`, async () => {
    const code = `
\`\`\`plantuml
digraph G {
node [shape="box"]
A [label="hello",shape="circle"]
A->B->C
A->C
label="Title"
}
\`\`\`
`

    await testRemarkPlugin.testPlugin({
      code,
    })
  })
})
