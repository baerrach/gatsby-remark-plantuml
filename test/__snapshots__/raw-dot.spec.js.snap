// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dot code renders 1`] = `
Object {
  "children": Array [
    Object {
      "children": undefined,
      "lang": undefined,
      "meta": null,
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 10,
          "offset": 109,
        },
        "indent": Array [
          1,
          1,
          1,
          1,
          1,
          1,
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 2,
          "offset": 1,
        },
      },
      "type": "html",
      "value": "<svg width=\\"90pt\\" height=\\"231pt\\" viewBox=\\"0.00 0.00 89.95 230.90\\" xmlns=\\"http://www.w3.org/2000/svg\\" xlink=\\"http://www.w3.org/1999/xlink\\" style=\\"max-width: 100%; height: auto;\\">
<g id=\\"graph0\\" class=\\"graph\\" transform=\\"scale(1 1) rotate(0) translate(4 226.9)\\">
<title>G</title>
<polygon fill=\\"white\\" stroke=\\"transparent\\" points=\\"-4,4 -4,-226.9 85.95,-226.9 85.95,4 -4,4\\"/>
<text text-anchor=\\"middle\\" x=\\"40.97\\" y=\\"-7.8\\" font-family=\\"Times New Roman,serif\\" font-size=\\"14.00\\">Title</text>
<!-- A -->
<g id=\\"node1\\" class=\\"node\\">
<title>A</title>
<ellipse fill=\\"none\\" stroke=\\"black\\" cx=\\"54\\" cy=\\"-194.95\\" rx=\\"27.9\\" ry=\\"27.9\\"/>
<text text-anchor=\\"middle\\" x=\\"54\\" y=\\"-191.25\\" font-family=\\"Times New Roman,serif\\" font-size=\\"14.00\\">hello</text>
</g>
<!-- B -->
<g id=\\"node2\\" class=\\"node\\">
<title>B</title>
<polygon fill=\\"none\\" stroke=\\"black\\" points=\\"54,-131 0,-131 0,-95 54,-95 54,-131\\"/>
<text text-anchor=\\"middle\\" x=\\"27\\" y=\\"-109.3\\" font-family=\\"Times New Roman,serif\\" font-size=\\"14.00\\">B</text>
</g>
<!-- A&#45;&gt;B -->
<g id=\\"edge1\\" class=\\"edge\\">
<title>A-&gt;B</title>
<path fill=\\"none\\" stroke=\\"black\\" d=\\"M45.31,-168.21C42.38,-159.55 39.11,-149.86 36.15,-141.1\\"/>
<polygon fill=\\"black\\" stroke=\\"black\\" points=\\"39.38,-139.71 32.86,-131.35 32.74,-141.95 39.38,-139.71\\"/>
</g>
<!-- C -->
<g id=\\"node3\\" class=\\"node\\">
<title>C</title>
<polygon fill=\\"none\\" stroke=\\"black\\" points=\\"81,-59 27,-59 27,-23 81,-23 81,-59\\"/>
<text text-anchor=\\"middle\\" x=\\"54\\" y=\\"-37.3\\" font-family=\\"Times New Roman,serif\\" font-size=\\"14.00\\">C</text>
</g>
<!-- A&#45;&gt;C -->
<g id=\\"edge3\\" class=\\"edge\\">
<title>A-&gt;C</title>
<path fill=\\"none\\" stroke=\\"black\\" d=\\"M58.99,-167.08C62.06,-147.18 65.06,-119.41 63,-95 62.28,-86.5 60.93,-77.31 59.49,-69.01\\"/>
<polygon fill=\\"black\\" stroke=\\"black\\" points=\\"62.91,-68.29 57.65,-59.09 56.03,-69.56 62.91,-68.29\\"/>
</g>
<!-- B&#45;&gt;C -->
<g id=\\"edge2\\" class=\\"edge\\">
<title>B-&gt;C</title>
<path fill=\\"none\\" stroke=\\"black\\" d=\\"M33.67,-94.7C36.71,-86.81 40.38,-77.3 43.76,-68.55\\"/>
<polygon fill=\\"black\\" stroke=\\"black\\" points=\\"47.07,-69.69 47.4,-59.1 40.54,-67.17 47.07,-69.69\\"/>
</g>
</g>
</svg>",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 1,
      "line": 11,
      "offset": 110,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;
